### ISSUE TITLE
Occassionally the Boost admin email will be sent before the UI is ready

### ISSUE THREAD

**Issue by PoeppingT** [awslabs/aws-saas-boost#83]
When running the installer, Cognito sends an email to the configured administrator email with a temporary password for use to login to the React app. In some cases (at least once for me) the email was sent long before the React app was finished provisioning, leading to a bad customer experience.

### Reproduction Steps

Run the installer. Watch your email closely, and immediately click the link once the email comes through. In some 

### What did you expect to happen?

I expected that the email would only show up in my inbox after the UI was ready for me to click the link.

### What actually happened?

The provided link sent me to a dead page. After ~2-3 minutes of refreshing, the UI finished provisioning and was reachable.

### Environment

  - **AWS Region  :** us-west-2
  - **AWS SaaS Boost Version :** main, HEAD https://github.com/awslabs/aws-saas-boost/commit/8bb3f795e96053e3838d3082a7ce84325b48695d
  - **Workload OS  (Linux or Windows) :** MacOS (Linux)

### Other

This may not be as easy to solve as it seems. We need Cognito to configure the user pool for access to the React app, and there is currently no hook to tell once the provisioning is done. We may need to add a new workflow to the CloudFormation template explicitly sending that email.

If after investigation this is found to be impossible, please close as won't fix.

--- 

This is :bug: Bug Report

**Comment by brtrvn** [comment]
If we had the saas-boost-web.yaml CloudFormation stack own the build of the React application (via a CodeBuild project), then we could use CloudFormation's DependsOn functionality to not create the `ConsoleAdminUser` resource prior to the CodeBuild project finishing and the CloudFront distribution being online (this is what takes the longest).

We'd change the installer to upload the source files for the React app to S3 just like it uploads the Lambda function packages prior to invoking create stack. CodeBuild would then grab the files from S3, run yarn, and sync the resulting React build to the S3 web hosting bucket.

This would also cushion users from having a full Node toolchain of the proper version installed and working locally in order to install SaaS Boost. Of course, if users want to hack on the web UI, they'd still need all the Node tools.
